llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1580,14 +1582,22 @@ static void computeKnownBitsFromOperator(const Operator *I,
                                                : SI->getTrueValue();
             IncDepth = Depth + 1;
           }
+        } else if (auto *IncPhi = dyn_cast<PHINode>(IncValue);
```

Comment:
We need a helper function `void breakSelfRecursion(Use &U, PHINode *PHI, Value*& ValOut, unsigned &DepthInOut, Instruction *&CtxIOut)` to avoid duplicate logic.

Reply:
I was thinking the same thing. Ill update.

Reply:
Done, let me know if you want me to break it out to an NFC